cli/cmd/project/clone.go

```diff
@@ -0,0 +1,153 @@
+package project
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	githttp "github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/rilldata/rill/cli/cmd/env"
+	"github.com/rilldata/rill/cli/pkg/cmdutil"
+	"github.com/rilldata/rill/cli/pkg/dotrillcloud"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/spf13/cobra"
+)
+
+func CloneCmd(ch *cmdutil.Helper) *cobra.Command {
+	var path string
+
+	cloneCmd := &cobra.Command{
+		Use:   "clone [<project-name>]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Clone Project",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			name := args[0]
+			if path == "" {
+				path = name
+			}
+
+			client, err := ch.Client()
+			if err != nil {
+				return err
+			}
+
+			// get project
+			res, err := client.GetProject(cmd.Context(), &adminv1.GetProjectRequest{OrganizationName: ch.Org, Name: name})
+			if err != nil {
+				return err
+			}
+
+			if res.Project.ArchiveAssetId != "" {
+				return fmt.Errorf("project is not connected to a git repository, please redeploy the project to connect it to a git repository")
+			}
+
+			// check if dir is empty
+			empty, err := isDirAbsentOrEmpty(path)
+			if err != nil {
+				return err
+			}
+			if !empty {
+				ok, err := cmdutil.ConfirmPrompt(fmt.Sprintf("Directory %q is not empty. Do you want to overwrite it?", path), "", false)
+				if err != nil {
+					return err
+				}
+				if !ok {
+					return fmt.Errorf("directory %q is not empty", path)
+				}
+			}
+
+			// create directory, the go-git SDK does not create the directory
+			err = recreateDir(path)
+			if err != nil {
+				return fmt.Errorf("failed to create directory %q: %w", path, err)
+			}
+
+			// get config
+			config, err := cmdutil.NewGitHelper(client, ch.Org, name, path).FetchGitConfig(cmd.Context())
+			if err != nil {
+				return err
+			}
+
+			// clone repository
+			_, err = git.PlainCloneContext(cmd.Context(), path, false, &git.CloneOptions{
+				URL:           config.Remote,
+				Auth:          &githttp.BasicAuth{Username: config.Username, Password: config.Password},
+				ReferenceName: plumbing.NewBranchReferenceName(config.DefaultBranch),
+				SingleBranch:  true,
+			})
```

Comment:
Why not have cloning as a function on the Git helper? Might be nice to keep Git logic in a single place.